### ISSUE TITLE
plays two pitches at once sometimes

### ISSUE THREAD

**Issue by JasonWoof** [wrschneider99/pitchpipe#1]
Thanks for posting this code! I love how simple the javascript is! This is exactly what I was hoping to find (the frequencies, and the oscillator code).

Sometimes when I press three ore more buttons it plays more than one tone at once. This is because of this line: ``currOscillator = undefined``. If osc is no longer the current oscillator then this line removes the (actually current) oscillator from that variable, so that this (now current) oscillator can't be stopped by the next button press.

To fix it, you could simply remove this line, or wrap a ``if (currOscillator === osc)`` check around it.

**Comment by wrschneider** [comment]
Hi, glad you found this useful.  I'd welcome a PR if you would like to offer a fix.

**Comment by wrschneider** [comment]
I think I understand the problem now.  If you push another button, you stop `curOscillator` but the `setTimeout` to stop it after a set duration is still left behind.  It is not a race condition or thread-safety issue, strictly speaking.  

Probably the simplest way to address this is to get rid of the assignment to `undefined` -  I think there is no impact to calling `stop` again on an oscillator that has already stopped.  

Also might be worth changing the `setTimeout` to use the duration parameter on `stop` directly.

**Comment by JasonWoof** [comment]
no, the problem is when the first one times out, it deletes the second one out of curOscillator without stopping it, so when the third one starts it can't stop the second one, because curOscillator no longer points to the currently paying oscillator (the second one).

I just made a pull request.
